pulsar-functions/java-examples/src/main/java/org/apache/pulsar/functions/api/examples/RecordFunction.java

```diff
@@ -37,6 +38,7 @@ public Record<String> process(String input, Context context) throws Exception {
         return context.<String>newOutputRecordBuilder()
                 .destinationTopic(publishTopic)
                 .value(output)
+                .schema(Schema.STRING)
```

Comment:
How can we prevent users from calling these methods:
- context.getCurrentRecord().ack()
- context.publish()
- context.newOutputMessage()
- context.getPulsarAdmin()

I understand that this will affect the functional boundaries provided by the sink itself. e.g. processing semantics.

Reply:
using that functions won't affect the result.
maybe your function has to publish a message also somewhere else, to log something

in this PIP we are simply running the function together with the Sink and removing the need of a intermediate topic.
that said the function can do whatever it wants

Reply:
I got it, thanks.